verticapy/tests/vDataFrame/test_vDF_correlation.py

```diff
@@ -35,27 +36,72 @@ def amazon_vd(base):
-            column="number", ts="date", by=["state"], p=5, show=False
+            ts="date",
+            column="number",
+            p=20,
+            by=["state"],
+            unit="month",
+            method="spearman",
+            show=False,
         )
-        assert result1["value"][0] == 1.0
-        assert result1["value"][1] == pytest.approx(0.515)
-        assert result1["value"][2] == pytest.approx(0.362)
-        assert result1["value"][3] == pytest.approx(0.208)
-        assert result1["value"][4] == pytest.approx(0.095)
-        assert result1["value"][5] == pytest.approx(0.006)
-
-        # making sure that vDataFrame.acf is the same
-        result1_1 = amazon_vd.acf(
-            column="number", ts="date", by=["state"], p=5, show=False
+
+        assert result1["value"][0] == pytest.approx(1)
+        assert result1["confidence"][0] == pytest.approx(0.0243968418)
+        assert result1.values["value"][10] == pytest.approx(0.48, abs=1e-2)
+        assert result1.values["confidence"][10] == pytest.approx(0.068, abs=1e-2)
+
+        # pearson method
+        result2 = amazon_vd.acf(
+            ts="date",
+            column="number",
+            by=["state"],
+            p=[1, 3, 6, 7],
+            unit="year",
+            method="pearson",
+            show=False,
+        )
+
+        assert result2["value"][0] == pytest.approx(1.0)
+        assert result2["confidence"][0] == pytest.approx(0.0243968418)
+        assert result2["value"][4] == pytest.approx(0.372, 1e-2)
+        assert result2["confidence"][4] == pytest.approx(0.040264769501212744, 1e-2)
+
+        # Autocorrelation Heatmap for each 'month' lag
+        result3 = amazon_vd.acf(
+            ts="date",
+            column="number",
+            by=["state"],
+            p=12,
+            unit="month",
+            method="pearson",
+            round_nb=3,
+            acf_type="heatmap",
+            show=False,
+        )
+
+        assert result3["index"][1].replace('"', "") == "lag_12_number"
+        assert result3["number"][1] == pytest.approx(0.701, 1e-2)
+        assert result3["index"][5].replace('"', "") == "lag_10_number"
+        assert result3["number"][5] == pytest.approx(0.308, 1e-2)
+
+        # Autocorrelation Line for each 'month' lag
+        result4 = amazon_vd.acf(
+            ts="date",
+            column="number",
+            by=["state"],
+            p=12,
+            unit="month",
+            method="pearson",
+            acf_type="line",
+            show=False,
         )
-        assert result1_1["value"][0] == result1["value"][0]
-        assert result1_1["value"][1] == result1["value"][1]
-        assert result1_1["value"][2] == result1["value"][2]
-        assert result1_1["value"][3] == result1["value"][3]
-        assert result1_1["value"][4] == result1["value"][4]
-        assert result1_1["value"][5] == result1["value"][5]
+
+        assert result4["value"][1] == pytest.approx(0.669, 1e-2)
```

Comment:
Imho, threshold of 1e-2 indicates that there is a big change in each run. Is there any randomness involved in acf and pacf?

Reply:
There is no randomness but if multiple dates are similar, the order can change and so it can change the output.